### PR TITLE
multi-handed item component

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
adds a component for items that require multiple hands to pick up.
if you're going to meme pr then at least do it properly.

also because this is one of those things everyone talks about needing yet isn't actually coded.
I even predicted it so it doesn't feel like dogass.
**Screenshots**

**Changelog**

no cl no fun

### CODE REVIEW

**Comment by mirrorcult** [comment]
isn't this just wielding

**Comment by moonheart08** [comment]
> isn't this just wielding

this is pickup requirements (i.e. you need two free hands to even so much as pick up a microwave)

**Comment by mirrorcult** [comment]
or so there isn't code duplication

**Comment by EmoGarbage404** [comment]
i imagine wielding is for entities with the option of wielding vs. not wielding. This is for things which absolutely require multiple hands. Thus, it would inevitably be mutually exclusive from wielding.

**Comment by GoodWheatley** [comment]
@moonheart08 once this is merged I can make picking up a microwave require 2 hands like you want

File: Content.Shared/Item/SharedMultiHandedItemSystem.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+namespace Content.Shared.Item;
+
+public abstract class SharedMultiHandedItemSystem : EntitySystem
+{
+    [Dependency] private readonly SharedHandsSystem _hands = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<MultiHandedItemComponent, GettingPickedUpAttemptEvent>((e,c,a) => OnAttemptPickup(e,c,a));
```
Comment (mirrorcult) [inline Content.Shared/Item/SharedMultiHandedItemSystem.cs:15]: wat

File: Content.Shared/Item/SharedMultiHandedItemSystem.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    protected virtual void OnEquipped(EntityUid uid, MultiHandedItemComponent component, GotEquippedHandEvent args)
+    {
+
+    }
+
+    protected virtual void OnUnequipped(EntityUid uid, MultiHandedItemComponent component, GotUnequippedHandEvent args)
+    {
+
+    }
```
Comment (mirrorcult) [inline Content.Shared/Item/SharedMultiHandedItemSystem.cs:29]: just make these abstract

File: Content.Shared/Item/SharedMultiHandedItemSystem.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    protected virtual bool OnAttemptPickup(EntityUid uid, MultiHandedItemComponent component, GettingPickedUpAttemptEvent args)
+    {
+        if (!TryComp<SharedHandsComponent>(args.User, out var hands))
+        {
+            args.Cancel();
+            return false;
+        }
+
+        var freeHands = _hands.EnumerateHands(args.User, hands).Where(x => x.IsEmpty);
+        if (freeHands.Count() < component.HandsNeeded)
```
Comment (mirrorcult) [inline Content.Shared/Item/SharedMultiHandedItemSystem.cs:40]: does handssystem not have a method for this already? i feel like wielding used it

**Review by mirrorcult — DISMISSED** [review]

**Pull request merged**